docs/server-elasticsearch-setup.md

```diff
@@ -0,0 +1,81 @@
+---
+id: server-elasticsearch-setup
+title: How to set up Elasticsearch for the Temporal Server
+sidebar_label: Set up Elasticsearch
+---
+
+## Overview
+
+You can integrate [Elasticsearch](https://www.elastic.co/elasticsearch/) with the Temporal Server to enhance Workflow search functionality.
+The [Server versions and dependencies](/docs/server-versions-and-dependencies) page lists compatible versions of dependencies.
+To set up Elasticsearch, there are two main Server configurations that need to be edited.
+
+- `persistence`
+- `kafka`
+
+Note that you can use the Temporal Server [development_es.yaml](https://github.com/temporalio/temporal/blob/master/config/development_es.yaml) file as working example.
+
+## Persistence
+
+Elasticsearch relies on storage to operate.
+Storage related configurations are defined within the `persistence` section of the configuration file.
+The Temporal Server needs to know where to connect to the Elasticsearch storage.
+
+To do this, first add the `advancedVisibilityStore: es-visibility` key value pair to the `persistence` section.
+
+```yaml
+persistence:
+  ...
+  advancedVisibilityStore: es-visibility
+```
+
+Next, define the ElasticSearch datastore connection information under the `es-visibility` key:
+
+```yaml
+persistence:
+  ...
+  datastores:
+    ...
+    es-visibility:
+      elasticsearch:
+        version: "v7"
+        url:
+          scheme: "http"
+          host: "127.0.0.1:9200"
+        indices:
+          visibility: temporal-visibility-dev
+```
+
+## Kafka
```

Comment:
kafka (as dependency for ES) is deprecated.

Reply:
Is it deprecated in Temporal Server v1.6 ?

If so, I suppose we can remove this whole section?

Reply:
@alexshtin

Reply:
In 1.6.0 config will still be needed. In 1.7.0 it will be completely removed (together with Kafka). If there is no urgency, I would remove this section from doc and wait till 1.7.0 release.

Reply:
Well, since this is a fairly big change - I think it is probably worth documenting the difference in dependencies and set up per versions. Once we get to v1.15.0 or something much farther along - then we can stop calling it out.